includes/downloads/recalculations.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Recalculation functions for downloads.
+ * @package     EDD
+ * @subpackage  Downloads
+ * @copyright   Copyright (c) 2022, Easy Digital Downloads, LLC
+ * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
+ * @since       3.1
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * The hook is registered here because it's specifically for the cron job,
+ * but it calls the general download recalculation function.
+ */
+add_action( 'edd_recalculate_download_sales_earnings_deferred', 'edd_recalculate_download_sales_earnings' );
+
+/**
+ * Handles scheduling the recalculation for a download.
+ *
+ * @since 3.1
+ * @param int $download_id
+ * @return void
+ */
+function edd_maybe_schedule_download_recalculation( $download_id ) {
+	$is_scheduled = wp_next_scheduled( 'edd_recalculate_download_sales_earnings_deferred', array( $download_id ) );
+	$bypass_cron  = apply_filters( 'edd_recalculate_bypass_cron', false );
+	if ( $is_scheduled && ! $bypass_cron ) {
+		edd_debug_log( 'Recalculation is already scheduled for ' . $download_id . ' at ' . edd_date_i18n( $is_scheduled, 'datetime' ) );
+		return;
+	}
+	if ( $bypass_cron || ( defined( 'EDD_DOING_TESTS' ) && EDD_DOING_TESTS ) ) {
+		edd_recalculate_download_sales_earnings( $download_id );
+		return;
+	}
+	edd_debug_log( 'Scheduling recalculation for ' . $download_id );
+	wp_schedule_single_event(
+		time() + ( 5 * MINUTE_IN_SECONDS ),
+		'edd_recalculate_download_sales_earnings_deferred',
+		array( $download_id )
+	);
+}
+
+add_action( 'edd_order_item_added', 'edd_recalculate_order_item_download', 10, 2 );
+add_action( 'edd_order_item_updated', 'edd_recalculate_order_item_download', 10, 3 );
+add_action( 'edd_order_item_deleted', 'edd_recalculate_order_item_download' );
+/**
+ * Attempts to schedule download sales and earnings when order items are changed.
+ *
+ * @since 3.1
+ * @param int $order_item_id                              The order item ID being added, updated, or deleted.
+ * @param array $data                                     The array of order item data being added or updated.
+ * @param bool\EDD\Orders\Order_Item $previous_order_item When updating an order, this is the original order item object.
+ * @return void
+ */
+function edd_recalculate_order_item_download( $order_item_id, $data = array(), $previous_order_item = false ) {
```

Comment:
In this function (and in the changes overall) can we give ourselves a little breathing room after the logical blocks? Having a comment for the next if statement come right after the closing bracket just makes it harder to read and follow since the brain logically groups them together instead of considering them separate 'tasks'.

Example:
```
	}
	// If the order item data being updated doesn't affect sales/earnings, recalculations do not need to be run.
	if ( $previous_order_item instanceof EDD\Orders\Order_Item ) {
```